packages/backend/src/core/SequenceProcessor.ts

```diff
@@ -0,0 +1,120 @@
+import { Logger, Retries, TaskQueue } from '@l2beat/common'
+import { assert } from '@l2beat/common/src/tools/assert'
+import { Knex } from 'knex'
+import { EventEmitter } from 'stream'
+
+import { SequenceProcessorRepository } from '../peripherals/database/SequenceProcessorRepository'
+
+export interface SequenceProcessorOpts {
+  id: string
+  logger: Logger
```

Comment:
Continuing my suggestion from earlier, logger and repository should be passed as regular function parameters anbd only primitive values and those config functions should go as options objects